### PR TITLE
[admin] auton integration merge

### PR DESCRIPTION
**Pull request by cgiger00**

### CODE REVIEW

File: jetson/nav/stateMachine.cpp [hunk 1]
```diff
@@ -166,6 +167,12 @@ NavState StateMachine::executeDrive() {
             return NavState::Done;
         }
     }
+    double distanceToWaypoint = estimateDistance(currentWaypoint.odom, mRover->odometry());
+    //TODO: refactor magic number (20 is maximum distance waypoint will be from target though)
+    if (mEnv->hasPostOneLocation() && distanceToWaypoint <= 20.0){
```
Comment (zwgold) [inline jetson/nav/stateMachine.cpp:172]: magic number, will change this

File: jetson/nav/search/searchStateMachine.cpp [hunk 2]
```diff
@@ -203,7 +202,7 @@ SearchFactory(const std::weak_ptr<StateMachine>& sm, SearchType type, const std:
     std::shared_ptr<SearchStateMachine> search = nullptr;
     switch (type) {
         case SearchType::FROM_PATH_FILE:
-            search = std::make_shared<SearchFromPathFile>(sm, roverConfig, "jetson/nav/search/spiral_search_points.txt");
+            search = std::make_shared<SearchFromPathFile>(sm, roverConfig, "jetson/nav/search/spiral_search_points_6m.txt");
```
[the remark this answers is not in the document]
Comment (ankithu) [reply jetson/nav/search/searchStateMachine.cpp:205]: oh shoot that's mb

File: jetson/nav/searchPointGeneration.py [hunk 3]
```diff
@@ -37,6 +37,20 @@ def generateSquareSpiral (points, distance):
     for i in range(0,points):
         coordinates.append( (coordinates[-1][0]+new_distance*directions[i%4][0], coordinates[-1][1]+new_distance*directions[i%4][1]) )
         new_distance += (i%2)*distance
+    #divide up each segment into intermediate segments
```
[the remark this answers is not in the document]
Comment (ankithu) [reply jetson/nav/searchPointGeneration.py:40]: yea so I just left it in and just returned the original coordinates so the code still exists if we ever want to go back to it

File: jetson/nav/search/searchStateMachine.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ankithu) [reply jetson/nav/search/searchStateMachine.cpp:205]: fixed in my pr

[the remark this answers is not in the document]
Comment (ankithu) [reply jetson/nav/search/searchStateMachine.cpp:205]: opening a pr to fix this

File: jetson/nav/searchPointGeneration.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (zwgold) [reply jetson/nav/searchPointGeneration.py:40]: Fine with me, just wanted other opinions on it

**Review by rbridges12 — APPROVED** [review]

**Review by zwgold — APPROVED** [review]
Waiting on a change to get approved to fix the magic number, but the `jetson/nav` code looks fine

**Review by qhdwight — APPROVED** [review]

**Comment by ankithu** [comment]
Once we get the pr with the testing changes merged in, looks good to merge to me

**Pull request merged**